lattices/src/fake.rs

```diff
@@ -32,15 +33,44 @@ impl<T> ConvertFrom<Fake<T>> for Fake<T> {
     }
 }
 
+impl<T, O> PartialOrd<Fake<O>> for Fake<T>
+where
+    T: PartialOrd<O>,
+{
+    fn partial_cmp(&self, other: &Fake<O>) -> Option<std::cmp::Ordering> {
+        self.0.partial_cmp(&other.0)
+    }
+}
```

Comment:
This one seems dubious
I guess though any compare is consistent with the merge if the merge just panics

Reply:
I think compare does need to be implemented for Fake to be useful at all, but it could also just panic immediately like merge does. I think I'll go with that implementation.